packages/client/components/modals/jobs/JobPost.tsx

```diff
@@ -0,0 +1,160 @@
+import {
+  Box,
+  Heading,
+  Image,
+  Text,
+  HStack,
+  VStack,
+  Stack,
+  Button,
+  Grid,
+} from '@chakra-ui/react';
+
+type CompanyInfo = {
+  name: string;
+  organization_description: string;
+  logo_url: string;
+};
+
+type Jobinfo = {
+  closeJobModal: () => void;
+  compensation: string;
+  description: string;
+  position: string;
+  companyInfo: CompanyInfo[];
+};
```

Comment:
Prefer to import the auto-generated type definitions from Supabase though so this enforces consistency in terms of how we are passing in the data to our DB

Example usage

```tsx
import { definitions } from '../../types/supabase';
```

Instead of using and defining another type for "CompanyInfo", use this instead `definitions['organizations']`

And instead of creating another "Jobinfo" type use `definitions['jobs']` instead

Though if you really need the `companyInfo: CompanyInfo[]` then I suggest this

```tsx
type JobInfo = {
  organizations: Array<definitions['organizations']>
} & definitions['jobs']
```

Reply:
I think this is mostly hardcoded atm and taken from the `JobCard` file as we're testing stuff. But you're right - using this will work charms going forward and we should indeed do that.